libs/ledger-live-common/src/onboarding/hooks/useOnboardingStatePolling.ts

```diff
@@ -49,11 +65,35 @@ export const useOnboardingStatePolling = ({
         next: (onboardingStatePollingResult: OnboardingStatePollingResult) => {
           if (onboardingStatePollingResult) {
             setFatalError(null);
-            setAllowedError(onboardingStatePollingResult.allowedError);
+            setAllowedError((prevAllowedError) => {
+              // Only updates if the new allowedError is different
+              if (
+                isDifferentError(
+                  prevAllowedError,
+                  onboardingStatePollingResult.allowedError
+                )
+              ) {
+                return onboardingStatePollingResult.allowedError;
+              }
+
+              return prevAllowedError;
+            });
 
             // Does not update the onboarding state if an allowed error occurred
             if (!onboardingStatePollingResult.allowedError) {
-              setOnboardingState(onboardingStatePollingResult.onboardingState);
+              setOnboardingState((prevOnboardingState) => {
+                // Only updates if the new onboardingState is different
+                if (
+                  isDifferentOnboardingState(
+                    prevOnboardingState,
+                    onboardingStatePollingResult.onboardingState
+                  )
+                ) {
+                  return onboardingStatePollingResult.onboardingState;
+                }
+
+                return prevOnboardingState;
```

Comment:
We can simplify the state function by moving all the logic to `isDifferentOnboardingState`
`return isDifferentOnboardingState(...)`

Reply:
Hi @mrbaguvix ! Thanks for your review :) I agree, i'll simplify with the state function by moving the logic in one function